code/game/gamemodes/dynamic/dynamic_unfavorable_situation.dm

```diff
@@ -11,9 +11,26 @@
 	var/static/list/unfavorable_random_events = list()
 	if (!length(unfavorable_random_events))
 		unfavorable_random_events = generate_unfavourable_events()
-	var/list/possible_heavies = list()
+	var/list/possible_heavies = generate_unfavourable_heavy_rulesets()
+	if (possible_heavies.len == 0)
```

Comment:
```suggestion
	if (!length(possible_heavies))
```

unlikely that we'll get a null instead of a list but let's be safe and clean since we're now reliant on another proc, doing it this way reads a bit cleaner to me anyways

Reply:
:-1: if we got null, we would want that to be a runtime